lib/libshare/smb.c

```diff
@@ -64,6 +64,21 @@ static boolean_t smb_available(void);
 
 static sa_fstype_t *smb_fstype;
 
+static void smb_strncpy(char *dst, char *src, size_t strlength)
+{
+	if (src == NULL || dst == NULL)
+		return;
+
+	strncpy(dst, src, strlength);
+	dst[strlength - 1] = '\0';
+}
```

Comment:
I'd suggest reworking the code to use `strlcpy()` which ensures the result in '\0' terminated.  You just need to make sure never to pass a NULL for `dst` or `src`.